docs/visual-basic/programming-guide/concepts/expression-trees/how-to-use-expression-trees-to-build-dynamic-queries.md

```diff
@@ -1,108 +1,140 @@
 ---
-description: "Learn more about: How to: Use Expression Trees to Build Dynamic Queries (Visual Basic)"
-title: "How to: Use Expression Trees to Build Dynamic Queries"
-ms.date: 07/20/2015
+title: "Querying based on runtime state (Visual Basic)"
+description: Describes various techniques your code can use to query dynamically depending on runtime state, by varying either LINQ method calls or the expression trees passed into those methods.
+ms.date: 14/02/2021
 ms.assetid: 16278787-7532-4b65-98b2-7a412406c4ee
 ---
-# How to: Use Expression Trees to Build Dynamic Queries (Visual Basic)
+# Querying based on runtime state (C#)
 
-In LINQ, expression trees are used to represent structured queries that target sources of data that implement <xref:System.Linq.IQueryable%601>. For example, the LINQ provider implements the <xref:System.Linq.IQueryable%601> interface for querying relational data stores. The Visual Basic compiler compiles queries that target such data sources into code that builds an expression tree at runtime. The query provider can then traverse the expression tree data structure and translate it into a query language appropriate for the data source.
+Consider code that defines an <xref:System.Linq.IQueryable> or an [IQueryable(Of T)](<xref:System.Linq.IQueryable%601>) against a data source:
 
-Expression trees are also used in LINQ to represent lambda expressions that are assigned to variables of type <xref:System.Linq.Expressions.Expression%601>.
+:::code language="vbnet" source="../../../../../samples/snippets/visualbasic/programming-guide/dynamic-linq-expression-trees/Program.vb" id="Initialize":::
```

Comment:
Adding to the first occurrence:  "vb" instead of "vbnet". (There are numerous occurrences)

```suggestion
:::code language="vb" source="../../../../../samples/snippets/visualbasic/programming-guide/dynamic-linq-expression-trees/Program.vb" id="Initialize":::
```